src/overlays/actors/ovl_Eff_Dust/z_eff_dust.c

```diff
@@ -22,28 +29,392 @@ const ActorInit Eff_Dust_InitVars = {
+            this->dx = 0.5f;
+            this->scalingFactor = 10.0f;
+            break;
+
+        case Effect_4:
+            EffDust_setUpdateFunc(this, EffDust_UpdateFunc_8099DFC0);
+            EffDust_setDrawFunc(this, EffDust_DrawFunc_8099E784);
+            this->actor.room = -1;
+            this->dx = 0.5f;
+            this->scalingFactor = 20.0f;
+            break;
+
+        default:
+            SystemArena_FreeDebug(this, "../z_eff_dust.c", 202);
+            break;
+    }
+
+    this->life = 0xA;
+}
+
+void EffDust_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EffDust_UpdateFunc_8099DB28(EffDust* this, GlobalContext* globalCtx) {
+    s16 theta;
+    s16 fi;
+
+    f32* distanceTraveled;
+
+    s32 i;
+    s32 j;
+
+    distanceTraveled = this->distanceTraveled;
+    for (i = 0; i < 0x40; i++) {
+        if ((*distanceTraveled) < 1.0f) {
+            *distanceTraveled += 0.05f;
+        }
+        distanceTraveled++;
+        // This won't match.
+        /*if (this->distanceTraveled[i] < 1.0f) {
+            this->distanceTraveled[i] += 0.05f;
+        }*/
+    }
+
+    for (j = 0; j < 3; j++) {
+        i = this->index & 0x3F;
+        if (this->distanceTraveled[i] >= 1.0f) {
+            // Spherical coordinate system.
+            fi = Rand_CenteredFloat(8192.0f);
+            theta = Rand_CenteredFloat(4096.0f);
+            this->initialPositions[i].x = -800.0f * Math_CosS(fi) * Math_CosS(theta);
+            this->initialPositions[i].y = -800.0f * Math_SinS(theta);
+            this->initialPositions[i].z = -800.0f * Math_SinS(fi) * Math_CosS(theta);
+            this->distanceTraveled[i] = 0.0f;
+            this->index += 1;
+        }
+    }
+}
+
+void EffDust_UpdateFunc_8099DD74(EffDust* this, GlobalContext* globalCtx) {
+    s16 theta;
+    s16 fi;
+
+    f32* distanceTraveled;
+
+    s32 i;
+    s32 j;
+
+    distanceTraveled = this->distanceTraveled;
+    for (i = 0; i < 0x40; i++) {
+        if ((*distanceTraveled) < 1.0f) {
+            *distanceTraveled += 0.03f;
+        }
+        distanceTraveled++;
+        // This won't match.
+        /*if (this->distanceTraveled[i] < 1.0f) {
+            this->distanceTraveled[i] += 0.03f;
+        }*/
+    }
+
+    for (j = 0; j < 2; j++) {
+        i = this->index & 0x3F;
+        if (this->distanceTraveled[i] >= 1.0f) {
+            // Spherical coordinate system.
+            fi = Rand_CenteredFloat(65536.0f);
+            theta = Rand_ZeroFloat(8192.0f);
+            this->initialPositions[i].x = 400.0f * Math_CosS(fi) * Math_CosS(theta);
+            this->initialPositions[i].y = 400.0f * Math_SinS(theta);
+            this->initialPositions[i].z = 400.0f * Math_SinS(fi) * Math_CosS(theta);
+            this->distanceTraveled[i] = 0.0f;
+            this->index += 1;
+        }
+    }
+}
+
+void EffDust_UpdateFunc_8099DFC0(EffDust* this, GlobalContext* globalCtx) {
+    s16 theta;
+
+    Player* player;
```

Comment:
```suggestion
    Player* player = PLAYER;
```
If possible, also keep the variables together without the extra spaces